drivers/sensor/bq2415x/bq2415x.c

```diff
@@ -0,0 +1,880 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (C) 2011-2013  Pali Rohár <pali@kernel.org>
```

Comment:
See this only now, this is not acceptable. You cannot take Linux Kernel code and change the license here.

Reply:
@filip-primozic can you confirm that this is the case? Did you take actual code from the Linux kernel? That would be a blocker as @jfischer-no mentions, since that code is GPLv2

Reply:
@jfischer-no , @carlescufi , yes this is the case. I took Linux kernel code as the base for the driver (since its the same chip and therefore the functionality is the same) and modified i2c functions as well as added bq2415x_init function to fit to Zephyr. I originally had GPLv2 license, but then changed it because when i pushed the code to github, compliance was failing. How do we go about this? I do not want to claim I wrote anything I did not write, that is why I left the Pali's name, in addition to my name. So, basically, is it possible to say "the base is from linux driver (with GPLv2 license), and some parts are modified (Zephyr requires Apache-2.0 license). This is my very first PR and would appreciate your help on this. Thanks a lot.

Reply:
Sorry for the delay @filip-primozic. The problem is that you cannot take code that is licensed under a particular license and just change the license in the header of the file. The authors of that file owns the copyright of that work, and they would have to agree to relicense it. That means that it is not possible to take Linux source code, modify it and then include it in Zephyr. You will need to write a clean-room implementation from scratch to be able to get it into Zephyr.